Reorg of the Usage section of the docs

melissawm opened the pull request:
# References and relevant issues
N/A

# Description
At the napari hackathon @ gloBIAS, we had the chance to discuss the docs organization with a few users and came up with ideas for potential improvements to our Usage section. We also discussed some changes to our API docs, but those will be handled separately.

~~This is a first very rough draft that is clearly incomplete, but I wanted to stash it somewhere so the work can continue.~~

EDIT: This is now ready for review. It only touches the first sections of the user guide but I think this is reasonable for now. This was discussed at the docs meeting this week (nov 20/21).

melissawm committed:
WIP

melissawm committed:
Lint

melissawm committed:
Undo changes to release notes

psobolewskiPhD commented:
This is really great! 😍 
Huge fan of breaking out the Layers into its own thing! This PR is the real closer to https://github.com/napari/docs/issues/42 😉 

One thing that stand out to me:
`Getting started > Open images` is empty -- just a placeholder for now? Maybe we should copy paste something into it and flag it as a followup?
(the quickstart `open an image` could use a spruce up too probably)

brisvag commented:
Love this work @melissawm! Much better than main :) Some thoughts:

- Did we come up with a better name for the `Features highlights` section in the end?
- I'd rename `napari tutorial` to something like `case studies` or `in depth examples` or `walkthroughs`. I'm also not sure if we want that there or lower down, but maybe we already discussed this?

DragaDoncila commented:
as mentioned in docs meeting, I love this reorg :heart: I dropped by because I noticed in the napari help menu we link to some of these guides, and I wanted to flag it in case we need to update any of those URLs. I don't think we do? Because it looks like it links to the file and we don't move the file. But just wanted to flag it and ask.

melissawm committed:
Previous commit

melissawm committed:
Changes from meeting

melissawm commented:
>  Getting started > Open images is empty -- just a placeholder for now? Maybe we should copy paste something into it and flag it as a followup?

Exactly! I can remove this from the ToC for now but we do think this is important, and was something users brought up during the hackathon. 

@DragaDoncila I will verify the help menu - good callout, I think we should be fine but we should check anyways. Worst case scenario we can also include redirects. 

EDIT: Only the Getting Started link needs to change to "https://napari.org/{VERSION}/getting_started/start_index.html", here's the PR: https://github.com/napari/napari/pull/8455

willingc commented:
@melissawm Looks great. Not sure if in or out of scope, but I think we can shorten a few headings:

- Shorten 'napari tutorials' to 'Tutorials'
- Shorten 'napari workshops' to 'Workshops'
- Shorten 'Layers User Guide' to 'Using layers'

<img width="628" height="638" alt="Screenshot 2025-11-29 at 2 18 30 PM" src="https://github.com/user-attachments/assets/63b716d9-5c40-4e33-801a-e9f26d795c3e" />

melissawm commented:
@willingc Thanks Carol! For tutorials and workshops I'm 💯 , for "Using layers" that's what we had before, and @TimMonko brought up something he saw on a talk which we agreed on, which is having the subject (layers in this case) be upfront in these titles so readers can skim better (Tim would you mind dropping a link here?)

However I'm not opposed to finding a shorter title if you have ideas! Maybe just "Layers guide"?

melissawm committed:
Shorten some titles

willingc commented:
Layers Guide would be great. Thanks @melissawm and @TimMonko

willingc reviewed (approved):
Thanks @melissawm

willingc commented:
@melissawm Maybe change the card to match Layers guide too. Same with Quickstart and Workshops.

<img width="1588" height="1062" alt="Screenshot 2025-11-30 at 8 32 44 AM" src="https://github.com/user-attachments/assets/bf84abfb-53ec-4a7b-9505-c643d6712b52" />

Czaki commented on docs/tutorials/cell_tracking.md:
```diff
@@ -1,3 +1,5 @@
+(tracking)=
```
Maybe `cell-tracking` To avoid future conflicts?

Czaki reviewed (commented):
I like this changes

melissawm committed:
Address reviews

melissawm commented:
Updated!

melissawm committed:
Emphasize layers as data

guiwitz commented:
Hi @melissawm. This looks great! BTW I created the docs locally using the pixi instructions and it worked really easily even though I'm not really a pixi user. I tried to go through the docs systematically and here are a few things I noticed:

- Plugins: Plugins are mentioned in multiple places but there's never a paragraph to explain what they are or a link to the actual "Finding and installing plugins" page. For example in the Quickstart they are mentioned multiple times but the links only send to the hub or actual plugins pages. Same for the "Getting Started" section. For example plugins are mentioned in the in "Install" section but just to say that depending on how you install napari, certain plugins might not work. I have the impression a new user would go through these docs and not really understand this. I think plugins are not just some kind of optional additional thing you can do, but actually essential, e.g. they are **required** to open most non-tiff microscopy images. So my suggestion would be: 
    - in the "Quickstart", add a section "Plugins" quickly explaining what they are and sending readers to the [actual plugin section](https://napari.org/stable/plugins/start_using_plugins/finding_and_installing_plugins.html#finding-and-installing-plugins)
    - in the "Getting Started -> Install napari" section add a specific part about installing plugins. I feel basic napari installation and plugin installation are tightly linked, so it would make sense to have it there. Again, no need to repeat content, but liking to the [plugin page](https://napari.org/stable/plugins/start_using_plugins/finding_and_installing_plugins.html#finding-and-installing-plugins) with minimum context would be sufficient.
- Beginners: I still think that at the moment the introductory material contains too much code. The Quickstart has code, the Tour of the napari viewer has code etc. I fear this will turn away beginners who don't regularly code. I understand the goal of not duplicating information (e.g. showing how to use the layer list interactively and programmatically) but I think it would be useful to have at least a no-code simplified intro. Just as a simple example: if a user just wants to visualize some data in 3D (for which napari is really good), currently the [2d/3d toggling](https://napari.org/stable/tutorials/fundamentals/viewer.html#toggle-2d-3d-button) relies on importing data with code, while one could just import the demo cells3d image, show the 2D version, show the button, show the 3d version.
- Open image: I'm happy to have a go at this if nobody else is working on it. Maybe in a new PR once this one is merged?

brisvag commented:
All great points!

> Open image: I'm happy to have a go at this if nobody else is working on it. Maybe in a new PR once this one is merged?

You can get started already if you want :wink:

melissawm commented:
Thank you so much @guiwitz ! Excellent points - I will try to address them in my next commit 😄

Czaki commented:
bump

melissawm committed:
Add release notes to fix CI failures

melissawm committed:
Improve quickstart, add plugins content

melissawm commented:
Hello, folks! I tried addressing all the comments above. It may not be perfect but I'd like to merge this for 0.7.0 and we can continue improving in the meantime. I don't know how to solve the codespell errors though - the release notes seem to have some kind of formatting error at the source? 

Anyway, @guiwitz if you can take a look soon that would be awesome. There is certainly room for improvements in all of these changes but at least we have a starting point.

Thanks!

psobolewskiPhD commented on docs/getting_started/quick_start.md:
```diff
@@ -14,7 +14,7 @@ kernelspec:
 (napari-quick-start)=
 
-# Quick start
+# Quickstart
```
My dictionary and my wife suggest this should be reverted to two words. cc: @napari/copy 
```suggestion
# Quick start
```

willingc replied:
@psobolewskiPhD Both versions are now acceptable. Quickstart has become the norm for technical guides, while quick start means beginning quickly. I'm fine with either though would lean toward one word.

melissawm replied:
Yeah I've been corrected in other projects where people made me go with the one-word version. I'd prefer keeping it !

psobolewskiPhD commented on docs/getting_started/quick_start.md:
The one linked here doesn't really work anymore as far as I can tell. Let's use the newer better one from Guillaume!
```suggestion
Alternatively, try [this plugin](https://napari-hub.org/plugins/napari-skimage.html) to use a GUI widget instead of the console.
```

psobolewskiPhD commented on docs/getting_started/installation.md:
```diff
@@ -0,0 +1,448 @@
+
+Choose one of the options below to install napari as a Python package.
+
+::::::\{tab-set}
```
This tab set doesn't appear to be working in the Circle preview at least:

<img width="1390" height="1020" alt="Image" src="https://github.com/user-attachments/assets/6db7eb70-e455-452f-995f-38be8a10af5d" />

melissawm replied:
Ugh, yes. This is a weird side effect of the linter, something's messed up in those configs. I'll fix, thanks!

melissawm replied:
Ugh, yes. This is a weird side effect of the linter, something's messed up in those configs. I'll fix, thanks!

psobolewskiPhD commented on docs/getting_started/installation.md:
```diff
@@ -0,0 +1,448 @@
+
+Click the tile below to jump to the installation instructions for your operating system.
+
+::::\{grid}
```
This grid isn't working in the Circle preview:

<img width="1390" height="1020" alt="Image" src="https://github.com/user-attachments/assets/d9df62ba-649d-4303-a709-802f7f00a605" />

psobolewskiPhD commented on docs/getting_started/launch_napari.md:
For followup: we should move this to the Troubleshooting page and link that page here.

melissawm replied:
Great catch! Created a new issue for this.

psobolewskiPhD commented on docs/getting_started/open_images.md:
```diff
@@ -0,0 +1 @@
+# Open images in napari
```
copy-paste from quick start, just so it's not empty. We should make a follow-up issue to improve on this.

````suggestion
# Open images in napari

napari natively supports tiff and many other formats supported by [skimage.io.imread](https://scikit-image.org/docs/dev/api/skimage.io.html) as input image file format.

Try with your own images or download [this ome tiff file](https://downloads.openmicroscopy.org/images/OME-TIFF/2016-06/MitoCheck/00001_01.ome.tiff).

Additional input file formats may be supported [by plugins](https://www.napari-hub.org/).
Try [napari-aicsimageio](https://www.napari-hub.org/plugins/napari-aicsimageio) if you have czi, lif, or nd2 files.

Once you have the proper plugin installed, use File > Open Files(s)
and select the image file, or simply drag and drop the image into napari.

For demo purpose, we will use a sample image that comes with napari.

### Option 1: Using the GUI to open an image

1. In napari, go to `File -> Open Sample -> napari built-ins -> Cell`

![Open sample image](../_static/images/open_image.png)

### Option 2: Using the napari console to open an image

1. Open napari IPython console

![IPython console](../_static/images/IPython.png)

2. Type

```python
from skimage import data

viewer.add_image(data.cell(), name='cell')
```

```{code-cell} ipython3
---
tags: [remove-input]
---
import napari
from napari.utils import nbscreenshot

viewer = napari.Viewer()

from skimage import data
viewer.add_image(data.cell(), name='cell')
nbscreenshot(viewer, alt_text="image of a single cell opened in napari viewer")
```

```{note}
Once you have a napari viewer open, you can also make a new `Image` layer from an image (or URL to an image) copied to your Clipboard using `File -> New Image from Clipboard` menu item (keybinding {kbd}`Command/Ctrl+N`).
```
````

psobolewskiPhD reviewed (changes requested):
This looks great! I left some minor comments, which arn't really blockers but nice to haves.

The real issue is the tiles and grids in getting_startedinstallation.md page not rendering right, at least in the circle previews.
It looks like the colon fencing (in place of backtick fencing) isn't working right, because other places look fine.

willingc commented on docs/usage.md:
@melissawm To keep things easy to scan and consistent across pages, I suggest simplifying the title of the card from "napari quick start" to "Quickstart".

DragaDoncila committed:
Update docs/getting_started/open_images.md

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

psobolewskiPhD commented on docs/getting_started/open_images.md:
```diff
@@ -0,0 +1,51 @@
+# Open images in napari
```
```suggestion
---
jupytext:
  formats: ipynb,md:myst
  text_representation:
    extension: .md
    format_name: myst
    format_version: 0.13
    jupytext_version: 1.11.5
kernelspec:
  display_name: Python 3 (ipykernel)
  language: python
  name: python3
---

# Open images in napari
```

Oops! This should fix the warning, i didn't realize there was a run code cell because the preview build doesn't build the code cells.

guiwitz commented:
Hi @melissawm, sorry didn't get to look at this earlier. The additions about plugin information in the two starting guides are great. I also like the simplified "no-code" example about 3D rendering, that's the way to go! I think once the issue with the grid is fixed, this should be merged, and further improvements can be done in future PRs.

melissawm committed:
Update docs/getting_started/quick_start.md

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

melissawm committed:
Update docs/getting_started/open_images.md

Co-authored-by: Peter Sobolewski <76622105+psobolewskiPhD@users.noreply.github.com>

melissawm committed:
Fix grid/cards

melissawm committed:
Re-title card

melissawm commented:
I think I got it all. Still trying to figure out what to do with the release notes that are causing the codespell error, but the cards and tabs are now working. Thanks!

TimMonko commented:
@melissawm I'm not sure what's up with the Codespell fail because it should be skipped

https://github.com/napari/docs/blob/ab4b01d1a8a2e06192a1239aefbb1348344b36eb/.codespellrc#L3

and in the release index PR thats what I said it was for in the commit: [`86d9c5f` (#838)](https://github.com/napari/docs/pull/838/commits/86d9c5f7d70bd465f0a378f8a076b3552bf5b9e9)

Should it be `./docs/release/index.md`? Not sure why it would have been working before though. Codepsell hasn't had a release for a year so its not like behavior should change

melissawm commented:
Ohh I see. let me investigate

melissawm committed:
Try to fix codespell failure

melissawm commented:
Wooo you were right @TimMonko , thanks!! 🙏🏻

TimMonko commented:
Maybe we just got really lucky for a few months that there wasn't a misspelled truncation in that file? LOL

TimMonko reviewed (approved):
Looks good to me! I think the remaining issues have been fixed!

psobolewskiPhD reviewed (approved):
let's go!

melissawm commented:
Gentle ping! This may help @guiwitz continue with the Open Images page 😄

melissawm committed:
Merge branch 'main' into usage-reorg

TimMonko commented:
You can just merge next time @melissawm . There's two approvals already 🫣

melissawm commented:
I hate self merging!! But it's noted 😁
